File: Part_2/Python/predictAnkleTorque/testPredictAnkleTorqueFunction.py
#!/usr/bin/env python
import predictAnkleTorque as pat
import matplotlib.pyplot as mplt
import os, sys
currentdir = os.path.dirname(os.path.realpath(__file__))
parentdir = os.path.dirname(currentdir)
sys.path.append(parentdir)
import helperFunctions as hf

# Test your code by running the following

def testPredictAnkleTorque():
	files = os.listdir('dataset')

	i = 0
	for filename in files:
		data_array = hf.load_csv_file('dataset/' + filename)
		
		ankleTorquePredictions = pat.predictAnkleTorque(data_array)
		
		mplt.figure(i+1)
		mplt.plot(data_array[:,hf.variable_name_to_column_index('ank_torque')], 'b')
		mplt.plot(ankleTorquePredictions,'r')
		mplt.draw()

		i += 1

	mplt.show()

if __name__ == "__main__":
	testPredictAnkleTorque()

File: Part_2/Matlab/predictAnkleTorque/testPredictAnkleTorqueFunction.m
%% Test your code by running the following
close all;
clear all;
testPredictAnkleTorque();

function testPredictAnkleTorque()
    files = dir('dataset');
    files = files(3:end,1);

    for i = 1:size(files,1)
        filename = files(i).name;
        data_table = readtable(strcat('dataset/',filename));
        
        ankleTorquePredictions = predictAnkleTorque(data_table);
        
        figure
        plot(data_table.ank_torque, 'b')
        hold on;
        plot(ankleTorquePredictions,'r')
    end
end